worker/worker.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var (
+	// workerID is the id assigned by the scheduler
+	// after registering on scheduler.
+	workerID string
+)
+
+func init() {
+	loadConfig()
+}
+
+// Entry point of the worker application.
+func main() {
+	go registerWorker()
```

Comment:
One thing to consider, since registerWorker can trigger os.Exit, is the signal handler / deregister may not trigger on error. For the current state of the program, I don't think this is a big deal, but something to be mindful of.

Reply:
all error handling (if the error is fatal) calls the deregister function first and then exits (by using the `fatal` function). However, if the error is an unexpected, uncatchable error, such as a nil pointer dereference etc. the deregister method will not be called. In that case, a heartbeat monitor/ping monitor would be required to monitor workers as discussed above